ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -992,17 +992,7 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
       return ServerErrorCode.Bad_Request;
     }
     if (!skipPartitionAndDiskAvailableCheck) {
-      // 2. check if partition exists on this node and that the store for this partition has been started
-      if (storageManager.getStore(partition) == null) {
-        if (partitionsInCurrentNode.contains(partition)) {
-          metrics.diskUnavailableError.inc();
-          return ServerErrorCode.Disk_Unavailable;
-        } else {
-          metrics.partitionUnknownError.inc();
-          return ServerErrorCode.Partition_Unknown;
-        }
-      }
-      // 3. ensure the disk for the partition/replica is available
+      // 2. ensure the disk for the partition/replica is available
       List<? extends ReplicaId> replicaIds = partition.getReplicaIds();
       for (ReplicaId replica : replicaIds) {
```

Comment:
Is it easy to expose an API: `partition.getLocalReplica()` ?  so that we don't need to iterate all.

Reply:
I guess the `partition` is logical concept which is not supposed to distinguish local/remote replicas. That's why there is no such a method available.